compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/InterceptorAnalysisTask.java

```diff
@@ -0,0 +1,44 @@
+package io.ballerina.stdlib.graphql.compiler;
+
+import io.ballerina.compiler.syntax.tree.ClassDefinitionNode;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.Token;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.stdlib.graphql.compiler.service.validator.InterceptorValidator;
+
+import static io.ballerina.stdlib.graphql.compiler.Utils.READONLY_SERVICE;
+import static io.ballerina.stdlib.graphql.compiler.Utils.hasCompilationErrors;
+
+/**
+ * Validates a Ballerina GraphQL Interceptors.
+ */
+public class InterceptorAnalysisTask implements AnalysisTask<SyntaxNodeAnalysisContext> {
+
+    public void perform(SyntaxNodeAnalysisContext context) {
+
+        if (hasCompilationErrors(context)) {
+            return;
+        }
+        ClassDefinitionNode classDefinitionNode = (ClassDefinitionNode) context.node();
+        NodeList<Token> tokens = classDefinitionNode.classTypeQualifiers();
+        if (tokens.isEmpty()) {
+            return;
+        }
+        if (!tokens.stream().allMatch(token -> READONLY_SERVICE.contains(token.text()))) {
+            return;
+        }
```

Comment:
This should be validated using the Semantic API. Syntax API validations are not trivial.

In this case, consider if a service is defined using `readonly service` with more than one space in between. Or even worse, `readonly` in one line, and the `service` is in the next line. We cannot check it correctly. But using the Semantics APIs, we can check the qualifiers of a service definition, so the syntax does not matter anymore. 

Check how the validation is done for a GraphQL service.